### DISCUSSION TITLE
How are carbon intensities for storage estimated?

### DISCUSSION THREAD

**Discussion by veqtrus** [electricitymaps/electricitymaps-contrib#2841] [Q&A]
Carbon intensities for storage (hydro and batteries) are (almost?) always calculated by Tomorrow. It would be nice to have some information about how this is done.

Do you take into account the energy losses during charging/discharging? What are the assumptions and sources?

Do you take into account when charging happens and whether it is dependent on the carbon intensity at the time? For example energy might be stored only when there is excess wind generation. In that case the carbon intensity of the source electricity is effectively only that of wind, even if gas plants might still be running.

**Comment by corradio** [comment]
Hi @veqtrus ,

There's broadly two ways to account for emissions of storage systems:
1. allocate emissions at charging time
2. allocate emissions at discharging time

We chose to allocate emissions at discharge time in order to make sure they are influencing a consumer's decision to consume electricity at a given time.
We assume 0 losses, meaning that the footprint of discharge is the same as the footprint of the zone at the time of charge.
In order to simplify things further, we assume that that the footprint at time of charge is the average yearly footprint, which is computed once a year.

This is the current state of implementation and of course this can evolve.

Olivier
